### ISSUE TITLE
Unable to build in windows

### ISSUE THREAD

**Issue by arjun-g** [alexa/apl-core-library#2]
Hi,

I tried to build the apl core in windows machine and facing the following error when i execute `ninja -j1`

```
P:\OpenSource\apl-core-library\build>ninja -j1
[0/1] Re-running CMake...
Using msvc
-- Configuring done
-- Generating done
-- Build files have been written to: P:/OpenSource/apl-core-library/build/googletest-download
ninja: no work to do.
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE)
WARNPlease install Doxygen to generate the documentation
-- Configuring done
-- Generating done
-- Build files have been written to: P:/OpenSource/apl-core-library/build
[5/240] Performing patch step for 'pegtl'
FAILED: pegtl-prefix/src/pegtl-stamp/pegtl-patch
cmd.exe /C "cd /D P:\OpenSource\apl-core-library\build\pegtl-prefix\src\pegtl && patch --binary -p1 < P:/OpenSource/apl-core-library/patches/pegtl.patch && "C:\Program Files\CMake\bin\cmake.exe" -E touch P:/OpenSource/apl-core-library/build/pegtl-prefix/src/pegtl-stamp/pegtl-patch"
patching file .gitignore
Hunk #1 FAILED at 1 (different line endings).
1 out of 1 hunk FAILED -- saving rejects to file .gitignore.rej
patching file include/tao/pegtl/analysis/analyze_cycles.hpp
Hunk #1 FAILED at 10 (different line endings).
Hunk #2 FAILED at 90 (different line endings).
2 out of 2 hunks FAILED -- saving rejects to file include/tao/pegtl/analysis/analyze_cycles.hpp.rej
patching file include/tao/pegtl/argv_input.hpp
Hunk #1 FAILED at 5 (different line endings).
Hunk #2 FAILED at 22 (different line endings).
2 out of 2 hunks FAILED -- saving rejects to file include/tao/pegtl/argv_input.hpp.rej
patching file include/tao/pegtl/contrib/tracer.hpp
Hunk #1 FAILED at 6 (different line endings).
Hunk #2 FAILED at 25 (different line endings).
Hunk #3 FAILED at 69 (different line endings).
Hunk #4 FAILED at 85 (different line endings).
Hunk #5 FAILED at 102 (different line endings).
Hunk #6 FAILED at 118 (different line endings).
Hunk #7 FAILED at 126 (different line endings).
Hunk #8 FAILED at 134 (different line endings).
Hunk #9 FAILED at 142 (different line endings).
Hunk #10 FAILED at 159 (different line endings).
Hunk #11 FAILED at 176 (different line endings).
Hunk #12 FAILED at 186 (different line endings).
Hunk #13 FAILED at 194 (different line endings).
Hunk #14 FAILED at 202 (different line endings).
Hunk #15 FAILED at 211 (different line endings).
Hunk #16 FAILED at 219 (different line endings).
Hunk #17 FAILED at 227 (different line endings).
Hunk #18 FAILED at 235 (different line endings).
18 out of 18 hunks FAILED -- saving rejects to file include/tao/pegtl/contrib/tracer.hpp.rej
patching file include/tao/pegtl/input_error.hpp
Hunk #1 FAILED at 5 (different line endings).
Hunk #2 FAILED at 35 (different line endings).
2 out of 2 hunks FAILED -- saving rejects to file include/tao/pegtl/input_error.hpp.rej
patching file include/tao/pegtl/internal/file_mapper_posix.hpp
Hunk #1 FAILED at 32 (different line endings).
1 out of 1 hunk FAILED -- saving rejects to file include/tao/pegtl/internal/file_mapper_posix.hpp.rej
patching file include/tao/pegtl/internal/file_mapper_win32.hpp
Hunk #1 FAILED at 58 (different line endings).
Hunk #2 FAILED at 81 (different line endings).
Hunk #3 FAILED at 93 (different line endings).
Hunk #4 FAILED at 143 (different line endings).
Hunk #5 FAILED at 164 (different line endings).
5 out of 5 hunks FAILED -- saving rejects to file include/tao/pegtl/internal/file_mapper_win32.hpp.rej
patching file include/tao/pegtl/internal/file_opener.hpp
Hunk #1 FAILED at 44 (different line endings).
Hunk #2 FAILED at 65 (different line endings).
2 out of 2 hunks FAILED -- saving rejects to file include/tao/pegtl/internal/file_opener.hpp.rej
patching file include/tao/pegtl/internal/file_reader.hpp
Hunk #1 FAILED at 32 (different line endings).
Hunk #2 FAILED at 70 (different line endings).
Hunk #3 FAILED at 90 (different line endings).
3 out of 3 hunks FAILED -- saving rejects to file include/tao/pegtl/internal/file_reader.hpp.rej
patching file include/tao/pegtl/position.hpp
Hunk #1 FAILED at 5 (different line endings).
Hunk #2 FAILED at 35 (different line endings).
2 out of 2 hunks FAILED -- saving rejects to file include/tao/pegtl/position.hpp.rej
ninja: build stopped: subcommand failed.
```

Am I missing anything ?

Thank you.

**Comment by cohenmickey** [comment]
Hi Arjun.

Can you check that your patch.exe in your path is the one shipped with git?  Gnu Patch for Windows is problematic.

-Mickey

**Comment by arjun-g** [comment]
Yes its using patch from git

```
P:\OpenSource\apl-core-library\build>where patch
C:\Program Files\Git\usr\bin\patch.exe
```

**Comment by arjun-g** [comment]
The error message I posted earlier had some wrong details. Updated it now.

**Comment by cohenmickey** [comment]
Thanks Arjun - another quick thing to try is to completely delete the build directory and re-try.

One thing that I did observe happening is that if ninja is run without the -j1 flag on the first run, it will fail, and subsequent runs from that directory could fail, even if the -j1 flag is given.  If that does not help, I or someone else will take a deeper look.

**Comment by arjun-g** [comment]
Hi @cohenmickey , tried that but no success. Tried applying git patch manually and removed those lines from `thirdparty.cmake`. Then got another error for which I have to add a line `#include <string>` to `aplcore/src/primitives/radii.cpp` . Now build succeeded :)

**Comment by bddufour** [comment]
Thanks for bringing this to our attention. Which compiler are you using? I want to try to reproduce this on our end.

**Comment by bddufour** [comment]
I was able to reproduce these issues using MSVC. The fix will be available in the next release.

**Comment by bddufour** [comment]
The fix for the windows build was released today as part of 1.3.3.
